src/main/java/com/summitsync/api/grouptemplate/GroupTemplateController.java

```diff
@@ -1,4 +1,65 @@
 package com.summitsync.api.grouptemplate;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController("/api/v1/grouptemplate")
 public class GroupTemplateController {
+
+    @Autowired
+    private GroupTemplateRepository repository;
+    private GroupTemplateMapper mapper;
+    @PostMapping
+    private ResponseEntity<GroupTemplateDTO> createTemplate(@RequestBody GroupTemplateDTO dto) {
+        repository.save(this.mapper.mapDtoToTemplate(dto));
+        return new ResponseEntity<>(dto, HttpStatus.CREATED);
+    }
+    @DeleteMapping
+    private ResponseEntity<GroupTemplateDTO> deleteTemplate(@PathVariable long id) {
+        Optional<GroupTemplate> template = repository.findById(id);
+        if (template.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        repository.deleteById(id);
+        return new ResponseEntity<>(this.mapper.mapTemplateToDto(template.get()), HttpStatus.OK);
+    }
+
+    @PutMapping
+    private ResponseEntity<GroupTemplateDTO> updateTemplate(@RequestBody GroupTemplateDTO dto) {
+        Optional<GroupTemplate> template = repository.findById(dto.getId());
+        if (template.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        GroupTemplate newTemplate = mapper.mapDtoToTemplate(dto);
+        this.repository.save(newTemplate);
+        return new ResponseEntity<>(this.mapper.mapTemplateToDto(newTemplate), HttpStatus.OK);
+    }
```

Comment:
Diese ganze Methode macht irgendwie kein Sinn für mich. Funktioniert die so? Du erstellt ja hier eine ganz neue GroupTemplate, und speicherst die. Das würde ja auch zu einer neuen ID in der Datenbank führen. Die alte template sollte hier ja durch die neue geupdated werden.